src/dscim/preprocessing/input_damages.py

```diff
@@ -717,20 +717,30 @@ def prep(ds, gcm=gcm):
 
 def coastal_inputs(
     version,
-    adapt_type,
-    vsl_valuation,
     path,
+    adapt_type,
+    vsl_valuation=None,
 ):
     try:
         d = xr.open_zarr(f"{path}/coastal_damages_{version}.zarr")
     except GroupNotFoundError:
         print(f"Zarr not found: {path}/coastal_damages_{version}.zarr")
         exit()
 
-    d = d.sel(adapt_type=adapt_type, vsl_valuation=vsl_valuation, drop=True)
-
-    d.to_zarr(
-        f"{path}/coastal_damages_{version}-{adapt_type}-{vsl_valuation}.zarr",
-        consolidated=True,
-        mode="w",
-    )
+    if "vsl_valuation" in d.coords:
```

Comment:
Note that with the new logic, there is a potential for the user to pass `vsl_valuation=None` and it also be a coord in the dataset --> this would throw an exception. Would be best to handle this. Probably add a clause inside this one. Something like:
```
if "vsl_valuation" in d.coords:
    if vsl_valuation == None:
        raise ValueError("vsl_valuation is a coordinate in xxxx dataset but is set to None. Please provide a value for vsl_valuation")
    else:
        # the rest of the code here.....